### PR TITLE
Adding sparse fieldset support

### LINKED ISSUE #11: Add sparse fieldset support.
http://jsonapi.org/format/#fetching-sparse-fieldsets

Should we accept the raw param?

Or a formatted list like:
```elixir
PostSerializer.format(model, conn, fields: [post: [:title, :published_at]])
```

### PR DESCRIPTION
**Pull request by green-arrow**
Resolves #11 (see that issue for additional details on the approach used here)

If you'd like, I can open another PR sometime to go through and update the README to reflect the recent additions of optional includes / sparse fieldsets. Not sure if you enjoy doing documentation or not :stuck_out_tongue:

### CODE REVIEW

**Comment by alanpeabody** [comment]
Thanks @green-arrow I will review tonight, add documentation and push up a new release, thanks!

File: lib/ja_serializer/builder/attribute.ex [hunk 1]
```diff
@@ -3,8 +3,24 @@ defmodule JaSerializer.Builder.Attribute do
 
   defstruct [:key, :value]
 
-  def build(%{model: model, serializer: serializer, conn: conn}) do
-    Enum.map apply(serializer,:attributes, [model, conn]), &do_build/1
+  def build(%{model: model, serializer: serializer, conn: conn} = context) do
```
Comment (alanpeabody) [inline lib/ja_serializer/builder/attribute.ex]: I think this is probably throwing compiler warnings, I don't think we need any of the map pattern matching here.

File: lib/ja_serializer/builder/attribute.ex [hunk 2]
```diff
@@ -3,8 +3,24 @@ defmodule JaSerializer.Builder.Attribute do
...
+  end
+
+  def fields_to_include(%{model: model, serializer: serializer, conn: conn} = context) do
+    attrs = apply(serializer,:attributes, [model, conn])
+    field_map = context[:opts][:fields] || %{}
+    attrs_to_include = Map.get(field_map, serializer.type)
+
+    cond do
+      attrs_to_include ->
+        include_list = String.split(attrs_to_include, ",") |> Enum.map(&(String.to_atom(&1)))
```
Comment (alanpeabody) [inline lib/ja_serializer/builder/attribute.ex]: could this just be `|> Enum.map(&String.to_atom/1)` ?

**Comment by alanpeabody** [comment]
This is super awesome, thanks! I left some minor nitpicks, but I am happy to merge and deal with my nitpicks myself, just let me know.

**Comment by green-arrow** [comment]
@alanpeabody no worries, those two things are fixed now :)

If you need help with updating the documentation, just let me know. I'd be happy to help with the elixir docs or the readme if you need it.

**Comment by alanpeabody** [comment]
Sure, feel free to open a documentation PR! I am going to add meta support then I think a 0.6.0 release will be ready.

**Pull request merged**